### PR TITLE
fix!: address NITs in PR #361

### PR DESCRIPTION
**Pull request by iajoiner**
Please be sure to look over the pull request guidelines here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/CONTRIBUTING.md#submit-pr.

# Please go through the following checklist
- [x] The PR title and commit messages adhere to guidelines here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/CONTRIBUTING.md. In particular `!` is used if and only if at least one breaking change has been introduced.
- [x] I have run the ci check script with `source scripts/run_ci_checks.sh`.

# Rationale for this change
Since I had auto-merge on NITs didn't get addressed. I promised to file a PR to resolve them today and hence I do now.

# What changes are included in this PR?
- remove `UnimplementedTestAccessor` which is no longer used
- remove `ProofPlan::is_empty` which is only used in `VerifiableQueryResult` by inlining
- make code more compact in `query_proof_test.rs` and `verifiable_query_result_test.rs`

# Are these changes tested?

Existing tests should pass.

### CODE REVIEW

**Comment by iajoiner** [comment]
Note that this is a breaking change since `ProofPlan::is_empty` was public

**Comment by iajoiner** [comment]
https://github.com/spaceandtimelabs/sxt-proof-of-sql/actions/runs/11786476589/job/32829840774?pr=367
I'm undoing the change to compactify the code base.

File: crates/proof-of-sql/src/sql/proof/verifiable_query_result.rs [hunk 1]
```diff
@@ -88,7 +88,11 @@ impl<CP: CommitmentEvaluationProof> VerifiableQueryResult<CP> {
         // have been rejected at the parsing stage.
 
         // handle the empty case
-        if expr.is_empty(accessor) {
+        let table_refs = expr.get_table_references();
+        if table_refs
+            .iter()
```
Comment (JayWhite2357) [inline crates/proof-of-sql/src/sql/proof/verifiable_query_result.rs:93]: NIT:
You could maybe consume `table_refs`.
```suggestion
            .into_iter()
```

**Review by JayWhite2357 — APPROVED** [review]
Approved, but with 1 small NIT.

**Review by JayWhite2357 — APPROVED** [review]

**Pull request merged**